app/models/deferred_offer_confirmation.rb

```diff
@@ -9,23 +9,18 @@ def courses_for_select
       @courses_for_select ||= offer.provider.courses
            .where(recruitment_cycle_year: RecruitmentCycleTimetable.current_year)
            .includes(:provider, :accredited_provider)
+           .distinct
            .order(:name)
     end
 
-    def course_options
-      @course_options ||= courses_for_select.map do |course|
-        [course.name_and_code, course.id]
-      end.unshift([nil, nil])
-    end
-
   private
 
     def no_raw_input
-      return if courses_for_select.size <= 20
+      return if courses_for_select.size < 20
       return if course_id.blank?
-      return if course_options.any? do |name, id|
-        course_id_raw == name && id == course_id.to_i
-      end
+
+      selected_course = courses_for_select.find_by(id: course_id)
+      return if selected_course && selected_course.name_and_code == course_id_raw
```

Comment:
`CandidateInterface::Concerns::FreeTextInputHelper`, could be moved out of the CandidateInterface to handle this.